unstructured_ingest/v2/processes/connectors/discord.py

```diff
@@ -0,0 +1,169 @@
+import datetime as dt
+from dataclasses import dataclass
+from typing import Any, Generator, List, Optional, Set
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.logger import logger
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+CONNECTOR_TYPE = "discord"
+
+
+class DiscordAccessConfig(AccessConfig):
+    token: str = Field(description="Discord API token")
+
+
+class DiscordConnectionConfig(ConnectionConfig):
+    access_config: Secret[DiscordAccessConfig]
+    channels: Optional[List[str]] = Field(
+        default=None, description="List of Discord channel IDs to process"
+    )
+
+
+class DiscordIndexerConfig(IndexerConfig):
+    recursive: bool = Field(
+        default=False, description="Recursively process child channels and messages"
+    )
+
+
+@dataclass
+class DiscordIndexer(Indexer):
+    connection_config: DiscordConnectionConfig
+    index_config: DiscordIndexerConfig
+
+    @requires_dependencies(["discord"], extras="discord")
+    def get_client(self):
+        import discord
+
+        intents = discord.Intents.default()
+        intents.message_content = True
+        return discord.Client(intents=intents)
+
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        client = self.get_client()
+        processed_channels: Set[str] = set()
+        channels_to_process: Set[str] = set(self.connection_config.channels or [])
+
+        while channels_to_process:
```

Comment:
Why does this need to be a `while` loop? I don't see it adding channels to be processed to warrant the while loop since that would dynamically change the list. As far as I can tell, this is simply iterating through the list of channels and getting the file data for it.

Reply:
This something maybe get list in migration that previously took the recursive flag into consideration?